census/ui_app/QuestionForm.jsx

```diff
@@ -186,20 +233,66 @@ const QuestionForm = React.createClass({
       if (this.getSchemaForId(q.id) === undefined) {
         console.warn('No schema defined for Question with id: ' + q.id);
       }
-      return (
-        <QuestionField ref={q.id}
-                       key={q.id}
-                       id={q.id}
-                       visibleProps={this.getVisiblePropsForId(q.id)}
-                       value={q.value}
-                       onChange={this.onFieldChange}
-                       label={this.getLabelForId(q.id)}
-                       position={this.getPositionForId(q.id)}
-                       instructions={this.getValueForId(q.id, 'description')}
-                       placeholder={this.getValueForId(q.id, 'placeholder')}>
-          {this.getValueForId(q.id, 'text')}
-        </QuestionField>
-      );
+      let questionNode = '';
+      switch (this.getValueForId(q.id, 'type')) {
```

Comment:
Yeah, I'm hate this repetition here. The default is currently just a fallback as I develop, likely to be replaced with something simpler once all question types have been defined. I like the idea of dynamically referencing type you have above. I'll try that out.